exa.py

```diff
@@ -125,6 +138,36 @@ def do(self, interp_state):
         interp_state.next_statement += 1
 
 
+class ADDI(MathStatement):
```

Comment:
I don't mind the cleanup changes being part of this PR, but please put them into separate commits.